File: C/Binary Tree - BFS/199. Binary Tree Right Side View.c
/* 199. Binary Tree Right Side View
Difficulty: Medium
Submission Link: https://leetcode.com/problems/binary-tree-right-side-view/submissions/1381336845/ */

void input(struct TreeNode* root, int* returnSize, int* ans, int now) {
    if (root == NULL) {
        return;
    }
    if (now > *returnSize) {
        ans[*returnSize] = root->val;
        *returnSize += 1;
    }
    input(root->right, returnSize, ans, now + 1);
    input(root->left, returnSize, ans, now + 1);
}

int* rightSideView(struct TreeNode* root, int* returnSize) {
    if (root == NULL) {
        *returnSize = 0;
        return NULL;
    }
    int* ans = malloc(sizeof(int) * 100);
    *returnSize = 0;
    input(root, returnSize, ans, 1);
    return ans;
}

File: C#/Binary Tree - BFS/199. Binary Tree Right Side View.cs
/* 199. Binary Tree Right Side View
Difficulty: Medium
Submission Link: https://leetcode.com/problems/binary-tree-right-side-view/submissions/1381336645/ */

public class Solution {
    public IList<int> RightSideView(TreeNode root) {
        List<int> level = new List<int>();
        if(root == null)
            return level;

        Queue<TreeNode> q = new Queue<TreeNode>();
        q.Enqueue(root);
        while (q.Count() != 0){
            int right = 0;
            int n = q.Count();
            for (var i = 0; i < n; i++){
                TreeNode node = q.Dequeue();
                right = node.val;
                if (node.left != null)
                    q.Enqueue(node.left);
                if (node.right != null)
                    q.Enqueue(node.right);
            }
            level.Add(right);
        }
        return level;
    }
}